src/Scouting.elm

```diff
@@ -1,141 +1,198 @@
-                    if model.isStarted then
-                        "Started"
-
-                    else
-                        "Startn't"
-                ]
-            ]
-        , Html.div []
-            [ Html.text <|
-                "Status:  name - "
-                    ++ model.scouterName
-                    ++ ", team - "
-                    ++ String.fromInt model.teamNum
-                    ++ ", match - "
-                    ++ String.fromInt model.matchNum
-                    ++ ", station - "
-                    ++ model.driverStation
-            ]
-        ]
+module Scouting exposing (Model, Msg, driverStations, init, main, update, view)
+
+import Browser
+import Html
+import Html.Attributes exposing (placeholder, style, value)
+import Html.Events exposing (onClick, onInput)
+import List.Extra exposing (elemIndex, getAt)
+import Maybe exposing (withDefault)
+import Maybe.Extra exposing (unwrap)
+import Tuple exposing (first, second)
+
+
+main : Program () Model Msg
+main =
+    Browser.sandbox
+        { init = init
+        , view = view
+        , update = update
+        }
+
+
+type Msg
+    = NameInput String
+    | TeamInput String
+    | MatchInput String
+    | Start
+
+
+type alias Model =
+    { scouter : String
+    , team : Maybe Int
+    , match : Maybe Int
+    , driverStation : String
+    , isStarted : Bool
+    }
+
+
+driverStations : List (List Int)
+driverStations =
+    [ [ 1690, 1574, 3339, 254, 2056, 1323 ]
+    , [ 118, 1577, 1024, 2056, 1690, 254 ]
+    , [ 1574, 3339, 1577, 1323, 1024, 118 ]
+    , [ 3339, 1574, 1577, 1024, 118, 2056 ]
+    ]
+
+
+init : Model
+init =
+    { scouter = ""
+    , team = Nothing
+    , match = Nothing
+    , driverStation = ""
+    , isStarted = False
+    }
+
+
+update : Msg -> Model -> Model
+update msg model =
+    case msg of
+        NameInput input ->
+            { model
+                | scouter = input
+            }
+
+        TeamInput input ->
+            { model
+                | team = String.toInt input
+                , driverStation = indexToName (String.toInt input) model.match
+            }
+
+        MatchInput input ->
+            { model
+                | match = String.toInt input
+                , driverStation = indexToName model.team (String.toInt input)
+            }
+
+        Start ->
+            { model
+                | isStarted = not model.isStarted
+                , driverStation = indexToName model.team model.match
+            }
+
+
+unwrapToString : Maybe Int -> String
+unwrapToString maybeInt =
+    unwrap "" String.fromInt maybeInt
+
+
+getMatchStations : Int -> List Int
+getMatchStations match =
+    -- == [team1, team2, team3, team4, team5, team6]
+    withDefault [ 0 ] <| getAt match driverStations
+
+
+stationIndex : Maybe Int -> Maybe Int -> String
+stationIndex team match =
+    -- team2 -> index = 2
+    unwrapToString << elemIndex (withDefault 0 team) <| getMatchStations (withDefault 0 match)

```

Comment:
Composition is _the_ way to reduce parentheses.
With some more composition, plus dropping `match` on both sides of the "equation":
```elm
stationIndex : Maybe Int -> Maybe Int -> String
stationIndex team =
    unwrapToString << elemIndex << withDefault 0 team << getMatchStations << withDefault 0
```
Notice how these two `withDefault`s are stuck there for no reason?
If we use a better design - one that makes sure we have valid data on the type level, we won't need them here 🙂.

Reply:
To clarify - when you have a function like
```elm
f: number -> number
f a =
    (+) 1 a
```
where `a` appears as the last parameter in the signature and the last argument of the definition, you can just omit it on both sides:
```elm
f : number -> number
f =
    (+) 1
```
just like
```
x + 1     = 2
x + 1 - 1 = 2 - 1
x         = 1
```

Reply:
Didn't work for me,
I'll maybe try it again later when the other things will work out.

Reply:
Remind me on Friday

Reply:
I've just fixed the parenthese - it should be ok now.